File: paddler_client_python/paddler_client/types/embedding.py
from pydantic import BaseModel, ConfigDict

from paddler_client.types.embedding_normalization_method import (
    EmbeddingNormalizationMethod,
)
from paddler_client.types.pooling_type import PoolingType


class Embedding(BaseModel):
    model_config = ConfigDict(frozen=True)

    embedding: list[float]
    normalization_method: EmbeddingNormalizationMethod
    pooling_type: PoolingType
    source_document_id: str

File: paddler_messaging/src/embedding.rs
use serde::Deserialize;
use serde::Serialize;

use crate::embedding_normalization_method::EmbeddingNormalizationMethod;
use crate::pooling_type::PoolingType;

#[derive(Debug, Deserialize, Serialize)]
#[serde(deny_unknown_fields)]
pub struct Embedding {
    pub embedding: Vec<f32>,
    pub normalization_method: EmbeddingNormalizationMethod,
    pub pooling_type: PoolingType,
    pub source_document_id: String,
}
